renku/core/management/datasets.py

```diff
@@ -262,29 +260,30 @@ def add_data_to_dataset(
 
         # commit all new data
         file_paths = {str(data['path']) for data in files if str(data['path'])}
-        self.repo.git.add(*(file_paths - set(ignored)))
+        new_files_count = len(file_paths) - len(ignored)
+        new_files = (file_paths - set(ignored))
 
-        if not self.repo.is_dirty():
-            return warning_message
+        if dataset.contains(new_files) is False and (new_files_count or force):
```

Comment:
Which (I would argue) that is exactly what we want - transactional add. Operation should not partially succeed since from users perspective is very strange to have partial success - since users might add something by accident which he does not want. We should force users to be explicit on the CLI and not silently succeeding at the operations and not letting them know what exactly happened.

Reply:
Sam, the following is obviously a bug:
```
[~/playground/tmp]$ renku dataset create a
Use the name "a" to refer to this dataset.
OK
[~/playground/tmp]$ renku dataset add a ~/some-file 
[~/playground/tmp]$ renku dataset add a ~/some-file ~/some-other-file 
[~/playground/tmp]$ git grep some-file
.gitattributes:data/a/some-file filter=lfs diff=lfs merge=lfs -text
.renku/datasets/f2960ea8-65a3-4a97-92e2-e87768d21f05/metadata.yml:  _id: file://blob/1f7d45364cbe7e7de19bfdfa01da7585351a9e5c/data/a/some-file
.renku/datasets/f2960ea8-65a3-4a97-92e2-e87768d21f05/metadata.yml:  _label: data/a/some-file@1f7d45364cbe7e7de19bfdfa01da7585351a9e5c
.renku/datasets/f2960ea8-65a3-4a97-92e2-e87768d21f05/metadata.yml:  name: some-file
.renku/datasets/f2960ea8-65a3-4a97-92e2-e87768d21f05/metadata.yml:  path: data/a/some-file
.renku/datasets/f2960ea8-65a3-4a97-92e2-e87768d21f05/metadata.yml:  url: file://../../some-file
[~/playground/tmp]$ git grep some-other-file
.gitattributes:data/a/some-other-file filter=lfs diff=lfs merge=lfs -text
[~/playground/tmp]$ git log --pretty=oneline
762b5de192477613185d8e90900a0f38c920b8b9 (HEAD -> master) renku dataset add a /home/mohammad/some-file /home/mohammad/some-other-file
bf50499e866daec0a1aa09170e36a422a6ba26ee renku dataset add a /home/mohammad/some-file
1f7d45364cbe7e7de19bfdfa01da7585351a9e5c renku dataset: committing 1 newly added files
e6dcaae7418a8c53f47ec412bae405032393a863 renku dataset create a
11b817913f9b58601b19aa1bc20d4f21ad196f06 renku init
```

It partially succeeds and it does not print an error message which I believe is the exact opposite of what you are aiming to do here.

Reply:
That is truly a bug! I will add a test case for it and fix it.

Thank you.